### ISSUE TITLE
ContextMenu: first item is focused when opening, but focus state is not shown

### ISSUE THREAD

**Issue by cschleiden** [OfficeDev/office-ui-fabric-react#934]
### Bug Report

- __Package version(s)__: master
- __Browser and OS versions__: *

#### Describe the issue:

When the context menu is opened, the first item automatically has focus, but since initially the `Fabric` element surrounding the `ContextMenu` has `isFocusVisible` set to false, the focus state is not shown. Only once you click or interact with the menu using cursor up/down is the focus state shown. 

I'm not sure what the correct fix is, either to not set focus to the first item, so that pressing Enter would not select it, until the user has explicitly selected an item with cursor up/down, or to highlight the focus item immediately. Might be related to #932

**Comment by dzearing** [comment]
1. Focus rectangles not showing up without keyboard interaction is by design.

2. There is a separate issue tracking that we should do some special casing for the mouse entry vs keyboard entry of menus. If you click on a menu button to expand the menu, we should keep focus on the button so that if you press enter, it hides the menu. However, if you use the keyboard to expand the menu button, moving focus in that case is desirable.

**Comment by cschleiden** [comment]
@dzearing but is it the correct behavior for a context menu to focus the first item without visual cue? In Windows Explorer, for example, when I open a context menu and hit [Enter], the context menu closes, the first item is **not** executed. 

In Chrome, just because I have it open right now, hitting [Enter] after opening a context menu also does **not** execute, but sets focus to the first item.

**Comment by cschleiden** [comment]
Additionally, for the focus highlight, when I tab via the keyboard to the element that opens the context menu, and then hit enter (in case of a button, like in the examples) shouldn't we show the focus in that case? 

I could understand if it's not shown if the menu is opened via mouse interaction.

**Comment by dzearing** [comment]
@cschleiden Yes that's correct behavior. Many things in ux will get focus without a visual cue. The focus rect design goals are:

1. Don't clutter the ux with unnecessary boxes whenever focus is set. It creates visually busy ux. Only show the focus rectangles when users are using the keyboard to navigate around the ux.

Focus rects add very little value unless you're navigating the ui with keyboard (e.g. think on a phone, or when purely using a mouse to navigate around.) An argument could be made for cross input scenarios; mouse click on menu, keyboard enter to contract the menu and it opens an item, but I design has opted to keep ux clean for the common cases.

In legacy windows explorer menus (e.g. native menus, not the ribbon), if you use the keyboard to navigate the menu, and you press enter to expand a menu, the first item is focused, such that if you press enter, it will execute that item. However if you expand the menu with a mouse click, focus is retained on the menu title and pressing enter contracts it.

**Comment by dzearing** [comment]
if you tab via keyboard to an element, focus rectangles SHOULD show up, for sure. Are you not seeing that?

**Comment by cschleiden** [comment]
Okay, didn't have a `<Fabric>` element surrounding my `Button` and the `ContextMenu`, so a new one was created once the menu was opened. That didn't know anything about my prior keyboard interaction, of course. 

So, for the next person who finds this: check your `<Fabric>` wrappers :)

Thanks for the clarification @dzearing.
